### PR TITLE
sci-libs/opencascade: bump to 7.6.1 / backport fixes from 7.6.1 to 7.6.0

### PR DESCRIPTION
**Pull request by waebbl**
[profiles/features/musl: mask sci-libs/opencascade](https://github.com/gentoo/gentoo/pull/24525/commits/c8a65d9b40f6a8f2b4a02c80230c495a3d14f40e)

Doesn't build with musl. Reported upstream, see
https://dev.opencascade.org/content/support-different-libc-implementations-linux

Bug: https://bugs.gentoo.org/832742
Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

[sci-libs/opencascade: bump to 7.6.1](https://github.com/gentoo/gentoo/commit/98bbecfc0d0806f0beab2f91353e58a66d3559e2)

Package is not yet compatible with ffmpeg-5, so we restrict to
<media-video/ffmpeg-5. Reported upstream.
Not compatible with musl libc implementation, so we explicitly
DEPEND on sys-libs/glibc.

Closes: https://bugs.gentoo.org/835040
Bug: https://bugs.gentoo.org/832742
Bug: https://bugs.gentoo.org/834422

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

[sci-libs/opencascade: backport fixes from v7.6.1](https://github.com/gentoo/gentoo/commit/1bbeecbc154d71607c22d97d1c9ba1a800c15075)

Package is not yet compatible with ffmpeg-5, so we restrict to
<media-video/ffmpeg-5. Reported upstream.
Not compatible with musl libc implementation, so we explicitly
DEPEND on sys-libs/glibc.

Closes: https://bugs.gentoo.org/832742
Closes: https://bugs.gentoo.org/834422

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

### CODE REVIEW

File: sci-libs/opencascade/opencascade-7.6.0-r3.ebuild [hunk 1]
```diff
@@ -44,7 +46,7 @@ RDEPEND="
 		dev-qt/qtwidgets:5
 		dev-qt/qtxml:5
 	)
-	ffmpeg? ( media-video/ffmpeg:= )
+	ffmpeg? ( <media-video/ffmpeg-5:= )
```
[the remark this answers is not in the document]
Comment (waebbl) [reply sci-libs/opencascade/opencascade-7.6.0-r3.ebuild:49]: For now, I don't see another way. I don't know the structure of ffmpeg at all and don't feel, I'm able to try and fix it myself. Haven't yet received a feedback on my upstream report. I revbumped it mainly for the addition of the glibc dependency. If it's not necessary to revbump for this, I'd be happy to revert the revbump.

[the remark this answers is not in the document]
Comment (waebbl) [reply sci-libs/opencascade/opencascade-7.6.0-r3.ebuild:49]: Just noticed, I missed to add the report: https://dev.opencascade.org/content/build-error-when-compiling-against-ffmpeg-5

File: sci-libs/opencascade/opencascade-7.6.0-r3.ebuild [hunk 2]
```diff
@@ -34,6 +35,7 @@ RDEPEND="
 	media-libs/fontconfig
 	media-libs/freetype:2
 	media-libs/ftgl
+	sys-libs/glibc
```
Comment (thesamesam) [inline sci-libs/opencascade/opencascade-7.6.0-r3.ebuild:38]: Note: unless it's a binary package, we should not depend on glibc, even if it fails to work on musl right now.

Depending on glibc makes it awkward to test in future. We should only depend on glibc if it's something we're truly stuck with (mainly binpkgs, but maybe some glibc-specific utility involving internals would work too).

File: sci-libs/opencascade/opencascade-7.6.0-r3.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (thesamesam) [reply sci-libs/opencascade/opencascade-7.6.0-r3.ebuild:49]: I think it's easier than avoiding a failed build part way through I guess.

File: sci-libs/opencascade/opencascade-7.6.0-r3.ebuild (hunk 2 above)
Comment (waebbl) [reply sci-libs/opencascade/opencascade-7.6.0-r3.ebuild:38]: I see. Sadly this was missed to mention when I asked on IRC. As it's already merged, I can revert this change on the next revbump and just maintain the mask for musl.

**Pull request closed without merging**